### ISSUE TITLE
Make it easier to change crypto policy

### ISSUE THREAD

**Issue by jlebon** [coreos/fedora-coreos-tracker#607]
This came out of discussions in https://discussion.fedoraproject.org/t/strong-crypto-settings-how-best-to-define-apply/22600.

This is a follow-up to https://github.com/coreos/fedora-coreos-tracker/issues/280 where we were able to ship the crypto policies, but split out `update-crypto-policies` into a subpackage which we don't ship because it pulls in Python.

However, this makes it cumbersome to change the policy. One has to change all the symlinks in `/etc/crypto-policies/back-ends` and the contents of `/etc/crypto-policies/state/current` via Ignition. An easier approach is to run `update-crypto-policies` from a container with `/etc/crypto-policies` mounted in but this requires restarting services (and it's also subject to version skew).

One suggestion in those issues above is to add FCCT sugar to set the policy. This would then update the symlinks. Long-term we could discuss with upstream whether there's any appetite for writing `update-crypto-policies` in a compiled language (that clearly might not appeal to them of course, but it's worth checking).

**Comment by travier** [comment]
Upstream code is at https://gitlab.com/redhat-crypto/fedora-crypto-policies.

**Comment by dustymabe** [comment]
We discussed this in the [community meeting today](https://lists.fedoraproject.org/archives/list/coreos@lists.fedoraproject.org/message/HFAD6TSELVURTAVGTJHT46HFXI4IA5FU/).

```
12:27:41      jlebon | #agreed we will recommend using Butane to set symlinks for now
                     | and approach upstream on whether there are any plans to rewrite
                     | the tool
```

So the action items now are:

- Add documentation with Butane snippets for configuring the crypto policy
- Approach upstream about a rewrite of [`update-crypto-policies`](https://gitlab.com/redhat-crypto/fedora-crypto-policies/-/blob/master/python/update-crypto-policies.py).

**Comment by fifofonix** [comment]
@dustymabe / @jelbon regarding documentation I am willing to do a PR for this in the next month but would like some guidance on how to shape.  My sense is that something centered on switching to FUTURE as an example of hardening + reference potential ways to further harden/loosen with user-based modifications might work.  And then make a caveat about FIPS being more complicated / not yet supported?

**Comment by jlebon** [comment]
> @dustymabe / @jlebon regarding documentation I am willing to do a PR for this in the next month but would like some guidance on how to shape. My sense is that something centered on switching to FUTURE as an example of hardening + reference potential ways to further harden/loosen with user-based modifications might work. And then make a caveat about FIPS being more complicated / not yet supported?

Yes, that sounds like a great start! We can link to the related tracker issue for the FIPS bit.
